integration/src/integration/java/pl/allegro/tech/hermes/integration/PublishingAvroTest.java

```diff
@@ -62,12 +65,69 @@ public void shouldPublishAvroAndConsumeJsonMessage() throws InterruptedException
     }
 
     @Test
+    public void shouldPublishAvroAndConsumeAvroMessage() throws InterruptedException, ExecutionException, TimeoutException, IOException {
+        // given
+        Topic topic = operations.buildTopic(topic()
+                .withName("publishAvroConsumeAvro.topic")
+                .withValidation(true)
+                .withMessageSchema(user.getSchemaAsString())
+                .withContentType(AVRO).build());
+        operations.createSubscription(topic, "subscription", HTTP_ENDPOINT_URL, ContentType.AVRO);
+
+        // when
+        Response response = publisher.publish("publishAvroConsumeAvro.topic", user.asBytes());
+
+        // then
+        assertThat(response.getStatus()).isEqualTo(CREATED.getStatusCode());
+        remoteService.waitUntilReceived(body -> {
+            AvroUser avroUser = AvroUser.create(user.getSchema(), body.getBytes());
+            assertThat(avroUser.getName()).isEqualTo("Bob");
+            assertThat(avroUser.getAge()).isEqualTo(50);
+            assertThat(avroUser.getFavoriteColor()).isEqualTo("blue");
+        });
+    }
+
+    @Test
+    public void shouldSendAvroAfterSubscriptionContentTypeChanged() throws InterruptedException, ExecutionException, TimeoutException, IOException {
+        // given
+        Topic topic = operations.buildTopic(topic()
+                .withName("publishAvroAfterTopicEditing.topic")
+                .withValidation(true)
+                .withMessageSchema(user.getSchemaAsString())
+                .withContentType(AVRO).build());
+        operations.createSubscription(topic, "subscription", HTTP_ENDPOINT_URL, JSON);
+
+        assertThat(publisher.publish("publishAvroAfterTopicEditing.topic", user.asBytes())).hasStatus(CREATED);
+        remoteService.waitUntilReceived(json -> assertThatJson(json).isEqualTo("{\"name\":\"Bob\",\"age\":50,\"favoriteColor\":\"blue\"}"));
```

Comment:
I think it would be nice to add to TestMessage ability to serialise itself as json string.